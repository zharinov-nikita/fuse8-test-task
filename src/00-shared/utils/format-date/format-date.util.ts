export function sharedUtilFormatDate(dateString: string) {
    const dateObject = new Date(dateString)

    const day = dateObject.getDate()
    const month = dateObject.getMonth() + 1
    const year = dateObject.getFullYear()

    const formattedDate = `${day < 10 ? '0' : ''}${day}.${
        month < 10 ? '0' : ''
    }${month}.${year}`

    return formattedDate
}
