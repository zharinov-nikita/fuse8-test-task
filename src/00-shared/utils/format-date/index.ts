export * from './format-date.util'
