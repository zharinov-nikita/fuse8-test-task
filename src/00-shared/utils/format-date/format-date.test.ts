import { describe, test, expect } from 'vitest'
import { sharedUtilFormatDate } from './format-date.util'

describe('sharedUtilFormatDate', () => {
    const data = '2020-01-05 13:42:18.823766'
    const result = '05.01.2020'
    test('case 1', () => {
        expect(sharedUtilFormatDate(data)).toBe(result)
        expect(sharedUtilFormatDate(data)).toMatchSnapshot()
    })
})
