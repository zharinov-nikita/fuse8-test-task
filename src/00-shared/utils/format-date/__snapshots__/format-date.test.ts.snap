// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`sharedUtilFormatDate > case 1 1`] = `"05.01.2020"`;

exports[`sharedUtilFormatDate > snapshot 1`] = `"05.01.2020"`;
